include/caffe/layer.hpp

```diff
@@ -293,6 +295,10 @@ class Layer {
     param_propagate_down_[param_id] = value;
   }
 
+  /**
+   * @brief Used by Net to give layers a pointer to their owning net.
+   */
+  void set_net(Net<Dtype>* net) { net_ = net; }
```

Comment:
Should there really be a setter or should this be an argument at construction? Are layers going to migrate between nets?